src/execution/execute.ts

```diff
@@ -328,9 +333,29 @@ export function experimentalExecuteIncrementally(
   return executeImpl(exeContext);
 }
 
+function subscribeToAbortSignal(exeContext: ExecutionContext): {
+  unsubscribeFromAbortSignal: () => void;
+} {
+  const onAbort = () => {
```

Comment:
On the other hand, I don't think you need any of this. The AbortSignal already has an `aborted` boolean property.  Since all we're doing is checking whether things are aborted inside completeValue, that can just check `aborted` directly, with no subscription needed. (Though if you take my suggestion from the top level of the review to always have an AbortController and hook up the user-provided one to it, then you'll still need the structure of subscribe/unsubscribe — it just won't be updating a boolean.)

Reply:
Decided to go with the option you suggested and create a new AbortController for each execution